openshift/console
# [release-4.9] Bug 2068203: Sample catalog is not displayed when one API call to the backend fails

**Fixes:**
https://bugzilla.redhat.com/show_bug.cgi?id=2068203

**Analysis / Root cause**: 


The reported issue contains a network traffic log that shows `504 Gateway Time-out` error for `/api/devfile/samples/?registry=https://registry.devfile.io`.

The sample catalog was stuck in the loading state when a network call fails. This could be also reproduced with the block URL feature in the chrome dev tools. See screenshots below.

**Solution Description**: 
This was already fixed in 4.10 by @gruselhaus.

This is a manual backport of #11579 / https://bugzilla.redhat.com/show_bug.cgi?id=2024262

**Screen shots / Gifs for design review**: 
Before (release-4.9, commit ``):


After (release-4.9 branch with this PR):



**Unit test coverage report**: 
Added by @gruselhaus 

**Test setup:**
1. Open the developer console > add > samples
2. Open the browser network inspector and filter for `devfile` or `sample`.
3. Refresh your browser tab to see that it works fine if the API works.
4. Right-click the network and block the URL
5. Refresh your browser tab to see that the page is not stuck and shows a warning that Devfiles could not be loaded.

**Browser conformance**:
